scripts/deploy.ts

```diff
@@ -3,287 +3,260 @@ import {
+            await executeContract(terra, wallet, network.builderUnlockAddress, {
+                "propose_new_owner": deployConfigs.teamUnlock.propose_new_owner
+            })
+        }
+    }
+}
+
+function checkAllocationAmount(allocations: Allocations[]) {
+    let sum = 0;
+
+    for (let builder of allocations) {
+        sum += parseInt(builder[1].amount);
     }
 
-    network.builderUnlockAddress = await deployContract(
-        terra,
-        wallet,
-        network.multisigAddress,
-        join(ARTIFACTS_PATH, 'builder_unlock.wasm'),
-        {
-            "owner": wallet.key.accAddress,
-            "astro_token": network.tokenAddress,
-            "max_allocations_amount": String(300_000_000_000000)
-        },
-        "Astroport Builder Unlocking Contract"
-    )
-
-    console.log("builderUnlockAddress", network.builderUnlockAddress)
-
-    let timestamp = (new Date().getTime() / 1000) | 0;
-    let seconds_in_day = 86400;
-    let duration = seconds_in_day * 365 * 3; // 3 years
-
-    let msg = Buffer.from(
-        JSON.stringify({
-            create_allocations: {
-                allocations: [
-                     [
-                        "terra14zees4lwrdds0em258axe7d3lqqj9n4v7saq7e",
-                        {
-                          amount: "1000000000",
-                          unlock_schedule: {
-                            start_time: timestamp,
-                            cliff: seconds_in_day,
-                            duration: duration,
-                          },
-                          proposed_receiver: undefined,
-                        },
-                      ],
-                    [
-                        "terra18wqkwdcz04upyg0eew3vyhepq9rgfl35aq6jw6",
-                        {
-                          amount: "1000000000",
-                          unlock_schedule: {
-                            start_time: timestamp,
-                            cliff: seconds_in_day,
-                            duration: duration,
-                          },
-                          proposed_receiver: undefined,
-                        },
-                    ],
-                ],
-            },
-        })
-    ).toString("base64");
-
-    let tx = await executeContract(
-      terra,
-      wallet,
-      network.tokenAddress,
-      {
-        send: {
-          contract: network.builderUnlockAddress,
-          amount: "2000000000",
-          msg
-        },
-      },
-      [],
-      `Setting ASTRO Unlocking schedules`
-    );
-
-    console.log(tx)
-
-    writeArtifact(network, terra.config.chainID)
+    if (sum != parseInt(deployConfigs.teamUnlock.initMsg.max_allocations_amount)) {
+        throw new Error(`Sum of allocations is ${sum}, but should be ${deployConfigs.teamUnlock.initMsg.max_allocations_amount}`);
+    }
+}
+
+async function create_allocations(terra: LocalTerra | LCDClient, wallet: Wallet, network: any, allocations: Allocations[]) {
```

Comment:
Let's just create allocations without storing any info

Reply:
We need to keep track of created allocations. If we don't and something happens in the middle you will have to rewrite allocation creations manually since you can not just restart script again. It will create new allocations for those who already have been created.

Reply:
Then we should create them at once as it was before